src/main/java/net/snowflake/ingest/utils/ParameterProvider.java

```diff
@@ -96,18 +95,23 @@ public class ParameterProvider {
    *
    * @param parameterOverrides Map of parameter name to value
    * @param props Properties from profile file
-   * @param isIcebergMode If the provided parameters need to be verified and modified to meet
-   *     Iceberg mode
    */
+  public ParameterProvider(Map<String, Object> parameterOverrides, Properties props) {
+    this.setParameterMap(parameterOverrides, props);
+  }
+
+  /** Constructor for tests */
   public ParameterProvider(
```

Comment:
why can tests not pass in a props with isIcebergMode set to true, instead of requiring a special constructor?

Reply:
that is, IMO we should remove this ctor.

Reply:
For test convenience, feel free to have a TestUtils.createParameterProvider(isIcebergMode) that ends up passing in a parameters or a props with isIcebergMode set to true.

Reply:
Delete ctor and create `createParameterProvider(isIcebergMode)` in TestUtils.